src/Catalogue.js

```diff
@@ -0,0 +1,23 @@
+class Catalogue {
+
+  constructor() {
+    this.products = []
+  }
+
+  add(p) {
+    this.products.push(p)
+  }
+
+  getAll() {
+    return this.products
+  }
+
+  getById(id) {
+    const result = this.products.filter(p => p.id === id)
```

Comment:
Aca me parece mas acertado usar el metodo `find()` en lugar de filter. mira lo que hace:  
Recibe una función con una condición y devuelve el primer elemento del array que cumpla con dicha condición, luego finaliza el método. En el caso de que no haya coincidencia devuelve Undefined.  
te queda una linea `this.product.find( p => p.id == id);`   
Otra cosa es el triple igual, pensa que en este momento te pueden pasar el id como un string y el triple igual no lo tomaria.

Reply:
Es verdad, ahí lo cambio. Igual el ID es un **String**

Reply:
Ahi vi, que es como una combinación de letras y numeros, en tal caso quiza si te conviene el `===` ya que entiendo que al haber letras siempre te van a pasar un string, yo creia que el id eran simples numeros como 1, 2, 3 etc..